joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala

```diff
@@ -2699,8 +2699,23 @@ class AstCreator(filename: String, javaParserAst: CompilationUnit, global: Globa
       logger.warn(s"Could not resolve the interface implemented by the lambda $expr. Type info may be missing.")
     }
 
-    // By definition, a functional interface will declare exactly one abstract method, so `find` is fine.
-    val maybeBoundMethod = maybeImplementedInterface.flatMap(_.getDeclaredMethods.asScala.find(_.isAbstract))
+    // While, in theory, there should only be a single abstract method defined by a functional interface, it turns
+    // out this is sort of not true for at least the java.util.Comparator interface, which re-declares the equals
+    // method which is marked `abstract` by JavaParser. The added check for parameter counts fixes this for the known
+    // example.
+    val maybeBoundMethod = maybeImplementedInterface.flatMap { interface =>
+      val abstractMethodsWithMatchingParamCount = interface.getDeclaredMethods.asScala
+        .filter(_.isAbstract)
+        .filter(_.getNumberOfParams == expr.getParameters.size())
+      if (abstractMethodsWithMatchingParamCount.size > 1) {
+        logger.warn(
+          s"Found multiple abstract methods matching param count for implementing lambda. Ignoring type information for stability"
+        )
+        None
```

Comment:
This is not required. According to https://docs.oracle.com/javase/8/docs/api/java/lang/FunctionalInterface.html it is allowed to have multiple abstract methods as long as only one of them is not already defined in `java.lang.Object`. So you can filter those away and we should be fine and not need this untyped fallback.